sdk/python/lib/pulumi/runtime/invoke.py

```diff
@@ -30,8 +29,15 @@
 # Note: We hit the default maximum protobuf size in practice when processing Kubernetes CRDs. If this setting ends up
 # causing problems, it should be possible to work around it with more intelligent resource chunking in the k8s provider.
 #
+# This import was raising an exception on Windows [2], and the bug [3] it was fixing does not appear to be present on
+# Windows, so only enable this option on other platforms.
+#
 # [1] https://github.com/protocolbuffers/protobuf/blob/0a59054c30e4f0ba10f10acfc1d7f3814c63e1a7/python/google/protobuf/pyext/message.cc#L2017-L2024
-SetAllowOversizeProtos(True)
+# [2] https://github.com/pulumi/pulumi/issues/3981
+# [3] https://github.com/pulumi/pulumi-kubernetes/issues/984
+if not sys.platform.startswith('win32'):
```

Comment:
Curious why this instead of `sys.platform == "win32"`?

Reply:
That idiom was recommended [in the docs](https://docs.python.org/3/library/sys.html#sys.platform). I assume `win32` would be fine as well, but wanted to be safe.